### PR TITLE
Extend prometheus jsonnet library with servicemonitor helpers

### PR DESCRIPTION
**Pull request by glrf**
Adds 3 new helper and extends the servicemonitor helper function a little to make it more convenient to use.

See https://github.com/appuio/component-openshift4-ingress/pull/45 for an example using these helpers

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [ ] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [x] Link this PR to related issues or PRs.

### CODE REVIEW

File: lib/prometheus.libsonnet [hunk 1]
```diff
@@ -121,8 +140,48 @@ local prometheusRule(name) = enable(kube._Object(api_version.monitoring, 'Promet
...
+ * By default this endpoint uses the service account token and the service-ca.
+ *
+ * \arg `serverName` The servername used to verify the target
+ * \return A service monitor endpoint with defaults for https and token authentication
+ */
+local serviceMonitorHttpsEndpoint(serverName) = {
+  bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+  interval: '30s',
+  scheme: 'https',
+  port: 'metrics',
```
Comment (simu) [inline lib/prometheus.libsonnet:170]: Not sure we want to set `port` here, as it seems to vary wildly between different services, from what I've seen. On the other hand it doesn't really hurt to have it either.

**Review by simu — APPROVED** [review]
LGTM overall

File: lib/prometheus.libsonnet (hunk 1 above)
Comment (glrf) [reply lib/prometheus.libsonnet:170]: Yes I was unsure about that too. My services all use `metrics` and I think two of yours do too, so I thought that might be an okeish default. Not sure if that's a representative selection though...

Comment (simu) [reply lib/prometheus.libsonnet:170]: I think it's a safe enough default, as I'd probably use `metrics` as the port name for a service which exposes metrics if I was writing the manifest myself.

**Pull request merged**